### PR TITLE
CI: Fix Serial Test Bug + Simplification + UX

### PR DESCRIPTION
**Pull request by kj4ezj**
## Change Description
From [AUTO-308](https://blockone.atlassian.net/browse/AUTO-308).

The CI system has been [running all tests](https://buildkite.com/EOSIO/eosio/builds/25248#fd2cee71-c326-4b2c-87fc-0da70bc5b5a6/211) for the serial test steps on `eos:develop-boxed`. The serial tests rely on container-level parallelization, so each step should only be running [one test](https://buildkite.com/EOSIO/eosio/builds/25191#8873d854-b763-4cc9-9952-11553b7fc7e3/189).

This pull request fixes the `develop-boxed` bug, simplifies some of the CI code, makes the pipeline easier to debug, and folds some log segments in order to focus on the parts of the build process that the customer cares most about.

The working directory was also changed from `/workdir` to `/eos` to support using the same container for multiple CI systems.

### See Also
- [Pull Request 9653](https://github.com/EOSIO/eos/pull/9653) -- `eos:develop-boxed`
- [Pull Request 9655](https://github.com/EOSIO/eos/pull/9655) -- `eos:blockvault`
- [Pull Request 9656](https://github.com/EOSIO/eos/pull/9656) -- `eos:develop`
- [Pull Request 9657](https://github.com/EOSIO/eos/pull/9657) -- `eos:release/2.0.x`
- [Pull Request 9658](https://github.com/EOSIO/eos/pull/9658) -- `eos:release/1.8.x`

## Change Type
- [ ] Documentation
- [ ] Stability bug fix
- [x] Other
CI
- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes
None.

## API Changes
- [ ] API Changes
None.

## Documentation Additions
- [ ] Documentation Additions
None.

### CODE REVIEW

**Comment by kj4ezj** [comment]
Note that the two unpinned test failures on this pull request are identical to those of last night's [build 4150](https://buildkite.com/EOSIO/eosio-build-unpinned/builds/4150) on the base branch, and this pull request is not intended to address those failures.

File: .cicd/test.sh [hunk 1]
```diff
@@ -5,13 +5,17 @@ set -eo pipefail
 # tests
 if [[ $(uname) == 'Darwin' ]]; then # macOS
     set +e # defer error handling to end
-    source ~/.bash_profile && "./$@"
+    [[ "$CI" == 'true' ]] && source ~/.bash_profile
```
Comment (johndebord) [inline .cicd/test.sh:8]: One thing to note is that macOS defaults to using zsh now. Not sure if that'll be an issue in the future

**Review by johndebord — APPROVED** [review]
LGTM.

File: .cicd/test.sh (hunk 1 above)
Comment (kj4ezj) [reply .cicd/test.sh:8]: Good point! As long as they don't break from Unix convention, [this line](https://github.com/EOSIO/eos/blob/d75a5f19e9560e85c306e46ba44b5ef554d777c4/.cicd/test.sh#L1) should direct macOS to run this script in BASH. However, I am concerned about them removing `bash` in the future.

**Pull request merged**